Makefile.dapper

```diff
@@ -28,6 +28,10 @@ shell: .dapper
 # This only lists targets accessible inside dapper (which are 99% of targets we use)
 targets:
 	@$(RUN_IN_DAPPER) make -pRrq : 2>/dev/null |\
-		grep -oP '^(?!Makefile.*)[-[:alnum:]]*(?=:)' | sort -u
+		grep -oP '^(?!Makefile.*)[-[:alnum:]]*(?=:)' | sort -u > /tmp/targets.txt;\
```

Comment:
Any particular reason to go through a temporary file here?

Reply:
It was simpler than figuring out how to do it in array inside the dapper call :)

It's inside the temporary container not affecting the host, but if it bothers you we can change it.

Reply:
Best I got it to output so far
```bash
$ make targets
sha256:f5c634c28eca812d104eebb8fd2f131be18e21dd1264e82e07dd07694dbb7977
[shipyard]$ trap chown -R 15900:15900 . exit
[shipyard]$ mkdir -p bin dist output
[shipyard]$ eval ${SCRIPTS_DIR}/targets.sh
[shipyard]$ /go/src/github.com/submariner-io/shipyard/scripts/shared/targets.sh
CODEOWNERS              
clean                    cleans everything (running clusters, generated files ...)
clean-clusters           removes running clusters
clean-generated          removes files we generated
cleanup                 
clusters                 creates KIND clusters that can then be used to deploy Submariner
deploy                   deploys Submariner on KIND clusters
deploy-latest            deploys the latest release
e2e                      executes the project's end to end testing on the deployed KIND clusters
gitlint                  validates the commits are valid
golangci-lint            validates Go code in the project
images                   builds all the container images for the project
lint                    
markdownlint             validates Markdown files in the project
nettest                 
post-mortem              prints a heap on information, to help in debugging failures on the KIND clusters
preload-images          
print-version           
prune-images             removes all Submariner-provided images and all untagged images
release                  uploads the project's images to a public repository
shellcheck              
unit                     executes the Go unit tests of the project
upgrade-e2e             
yamllint                 validates YAML files in the project
[shipyard]$ eval ${SCRIPTS_DIR}/targets.sh
```

Reply:
Oh, I wasn’t thinking as far as that — just

```suggestion
		for target in $((grep -oP '^(?!Makefile.*)[-[:alnum:]]*(?=:)' | sort -u); do\
```

Reply:
The pipe isn’t obvious, `make` runs, its output is piped to the `for` and consumed by the `grep` in the command substitution.

Reply:
Yes, one does not simply pipe output from make to a for and expect it to still run in dapper.

Reply:
Has to be a script run inside dapper, otherwise some of it runs inside and some outside (unless we want multiple dapper calls which would be not very efficient).

Reply:
Right, because we need to search for stuff in the Makefiles from the base image too.